### PR TITLE
comby: init at 0.12.1

### PR DESCRIPTION
**Pull request by Br1ght0ne**
###### Motivation for this change
https://github.com/comby-tools/comby/releases/tag/0.12.1

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/nixos/nixpkgs/76828)

### CODE REVIEW

File: pkgs/development/ocaml-modules/hack_parallel/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  src = fetchFromGitHub {
+    owner = "rvantonder";
+    repo = "hack-parallel";
+    rev = "v${version}";
+    sha256 = "051hbnj4lvvfdb3vs14apl7kwpdn9rbalnllfdc4xnhv3rzzxsyk";
+  };
+
+  buildInputs = [ core ppx_deriving ppxlib sexplib ];
+
+  patches = [ ./dune-static-libs.patch ];
```
Comment (yegortimoshenko) [inline pkgs/development/ocaml-modules/hack_parallel/default.nix]: Nitpick: technically it's a diff, not a patch :) Compare:

- https://github.com/NixOS/nixpkgs/commit/228627d50e1be526aa87a97c282ecf9dcbacc6eb.diff
- https://github.com/NixOS/nixpkgs/commit/228627d50e1be526aa87a97c282ecf9dcbacc6eb.patch.

**Review by lukateras — COMMENTED** [review]
Doesn't build:

```
(cd _build/default && .ppx/3a48ee47a0ff6f2bd43558b49bc4d853/ppx.exe --conditional --cookie 'library-name="server_types"' -o lib/server/server_types.pp.ml --impl lib/server/server_types.ml --dump-ast)
.ppx/3a48ee47a0ff6f2bd43558b49bc4d853/ppx.exe: unknown option '--conditional'.
builder for '/nix/store/y0wq6bdvfgywg2hcna23ix6x5g7q620f-ocaml4.07.1-comby-0.12.1.drv' failed with 
exit code 1
error: build of '/nix/store/y0wq6bdvfgywg2hcna23ix6x5g7q620f-ocaml4.07.1-comby-0.12.1.drv' failed
```

**Comment by yegortimoshenko** [comment]
@GrahamcOfBorg build comby

**Comment by JohnAZoidberg** [comment]
Still interested!

**Comment by kvtb** [comment]
There is upstream interest to get it packed in `nixpkgs` (likely the author is Nix/Darwin user)
https://github.com/comby-tools/comby/issues/142

**Comment by JohnAZoidberg** [comment]
I rebased it onto master and fixed a few things:

- Rename the patch file to diff
- Enable `useDune2 = true` because that's apparently required now.

And because comby depends on a few custom dependencies we need to either push upstream to upstream their forked dependency changes or patch/package those (see https://github.com/comby-tools/comby/blob/master/comby.opam):

- [x] mparser-comby: I temporarily added a new file for this
- [ ] patdiff: Requires a different branch that doesn't cleanly merge with upstream, so I can't create a patch file (https://github.com/rvantonder/patdiff.git#0.13.0-patch-compatible-diffs)
- [ ] hack_parallel: Not sure if this is the correct version
- [ ] bisect_ppx: Not sure if this causes issues
- [ ] camlzip: Not sure if this causes issues

**Comment by kvtb** [comment]
`1.0.0` is released, `hack_parallel` can be trivially updated, but `comby` is tricky

**Comment by dywedir** [comment]
Thanks!
Closing the PR, please see https://github.com/NixOS/nixpkgs/pull/123986, https://github.com/NixOS/nixpkgs/pull/139408

**Pull request closed without merging**